pkg/internal/controller/controller.go

```diff
@@ -167,6 +168,9 @@ func (c *Controller) Start(stop <-chan struct{}) error {
 			}
 		}
 
+		// All the watches have been started, we can reset the local slice.
```

Comment:
I'd add more details here as to why it's bad to hold on to the watchDescriptions

Reply:
Done!